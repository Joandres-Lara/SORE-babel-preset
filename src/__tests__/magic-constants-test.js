const babel = require('@babel/core');
const magicConstantsPlugin = require('../plugins/magic-constants');

const TEST_TRANSFORM_CODE = `
console.log([__TEST__,__DEV__]);
`;

test('insert __TEST__ expresion', () => {
 const { code } = babel.transform(TEST_TRANSFORM_CODE, { plugins: [magicConstantsPlugin] });
 expect(code).toMatchSnapshot();
});
