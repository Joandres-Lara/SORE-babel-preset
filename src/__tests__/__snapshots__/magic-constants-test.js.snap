exports[`insert __TEST__ expresion 1`] = `
"const __TEST__ = process.env.NODE_ENV === \\"test\\" || process.env.NODE_ENV === \\"testing\\";

const __DEV__ = process.env.NODE_ENV !== \\"production\\";

console.log([__TEST__, __DEV__]);"
`;
